ansible/plugins/callback/datadog_callback.py

```diff
@@ -0,0 +1,326 @@
+            self.disabled = False
+            print('Datadog callback disabled in "check mode".  ')
+        else:
+            self.disabled = False
+            # Set logger level - datadog api and urllib3
+            for log_name in ['requests.packages.urllib3', 'datadog.api']:
+                self._set_logger_level(log_name)
+
+        self._playbook_name = None
+        self._start_time = time.time()
+        self._options = None
+        if cli:
+            self._options = cli.options
+
+        extra_vars_file = self._options.extra_vars[1][1:]
+
+        with open(extra_vars_file, 'r') as stream:
+            try:
+                self.extra_vars = yaml.load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+        # self.playbook is set in the `v2_playbook_on_start` callback method
+        self.playbook = None
+        # self.play is set in the `playbook_on_play_start` callback method
+        self.play = None
+
+    # Set logger level
+    def _set_logger_level(self, name, level=logging.WARNING):
+        try:
+            log = logging.getLogger(name)
+            log.setLevel(level)
+            log.propagate = False
+        except Exception as e:
+            # We don't want Ansible to fail on an API error
+            print("Couldn't get logger - %s" % name)
+            print(e)
+
+    def _get_hostname(self,host):
+         try:
+            if host is not None:
+                hostname = self.hostvars[str(host)]['hostname']
+                return str(hostname) + "." + self.extra_vars['internal_domain_name']
+        except:
+            return host.split(".")[0]
+
+    def _send_event(self, title, alert_type=None, text=None, tags=None, host=None, event_type=None, event_object=None):
+        if tags is None:
+            tags = []
+        tags.extend(self.default_tags)
+        priority = 'normal' if alert_type == 'error' else 'low'
+        host = self._get_hostname(host)
+        
+        try:
+            datadog.api.Event.create(
+                title=title,
+                text=text,
+                alert_type=alert_type,
+                priority=priority,
+                tags=tags,
+                host=host,
+                source_type_name='ansible',
+                event_type=event_type,
+                event_object=event_object,
+            )
+        except Exception as e:
+            print('Couldn\'t send event "{0}" to Datadog'.format(title))
+            print(e)
+
+    # Send event, aggregated with other task-level events from the same host
+    def send_task_event(self, title, alert_type='info', text='', tags=None, host=None):
+        if getattr(self, 'play', None):
+            if tags is None:
+                tags = []
+            tags.append('play:{0}'.format(self.play.name))
+        self._send_event(
+            title,
+            alert_type=alert_type,
+            text=text,
+            tags=tags,
+            host=host,
```

Comment:
should this `host` also have the same treatment as the others? 

Also for a few others below

Reply:
All the functions used send_event and send_metrics, So I have modified only those two.

Reply:
Offlined with @NitigyaS. I just misunderstood the previous comment.